### PR TITLE
[KYUUBI #2024] Hive Backend Engine - ProcBuilder for HiveEngine

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

Add ProcBuilder for HiveEngine.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala [hunk 1]
```diff
@@ -23,5 +23,5 @@ package org.apache.kyuubi.engine
 object EngineType extends Enumeration {
   type EngineType = Value
 
-  val SPARK_SQL, FLINK_SQL, TRINO = Value
+  val SPARK_SQL, FLINK_SQL, TRINO, HIVE_SQL = Value
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala:26]: I think we should use `HIVE` here, for Flink and Spark, they provide user interfaces not only SQL, for Trino and Hive, only SQL, `HIVE_SQL` is quite redundant.

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala:26]: Oh, I see we already named the module `kyuubi-hive-sql-engine`, `kyuubi-trino-sql-engine` looks abnormal then...

File: externals/kyuubi-hive-sql-engine/bin/hive-sql-engine.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  HIVE_CLIENT_JAR="$HIVE_ENGINE_JAR"
+  HIVE_CLIENT_JARS_DIR="$HIVE_ENGINE_HOME/jars"
+else
+  echo "\nHIVE_ENGINE_HOME $HIVE_ENGINE_HOME doesn't match production directory, assuming in development environment..."
+  HIVE_CLIENT_JAR=$(find $HIVE_ENGINE_HOME/target -regex '.*/kyuubi-hive-sql-engine_.*.jar$' | grep -v '\-sources.jar$' | grep -v '\-javadoc.jar$' | grep -v '\-tests.jar$')
+  HIVE_CLIENT_JARS_DIR=$(find $HIVE_ENGINE_HOME/target -regex '.*/jars')
+fi
+
+HIVE_CLIENT_CLASSPATH="$HIVE_CLIENT_JARS_DIR/*"
+FULL_CLASSPATH="$HIVE_CLIENT_CLASSPATH:$HIVE_CLIENT_JAR:$HADOOP_CONF_DIR:$HIVE_CONF_DIR"
```
Comment (pan3793) [inline externals/kyuubi-hive-sql-engine/bin/hive-sql-engine.sh:35]: Do we need consider `YARN_CONF_DIR`?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala (hunk 1 above)
Comment (zhaomin1423) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala:26]: So, whether to rename them anywhere or keep them.

File: externals/kyuubi-hive-sql-engine/bin/hive-sql-engine.sh (hunk 2 above)
Comment (zhaomin1423) [reply externals/kyuubi-hive-sql-engine/bin/hive-sql-engine.sh]: Thanks, added.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala:26]: cc @yaooqinn

**Comment by yaooqinn** [comment]
> Thanks, merging to master

**Pull request closed without merging**